lib/init.js

```diff
@@ -78,7 +78,12 @@ function onInit() {
       log('debug', 'sites index populated');
     })
     .catch(err => {
-      throw err;
+      log('fatal', `${err.message}`, {
+        stack: err.stack
+      });
+      // A lot of platform functionality assumes the presence of this
+      // index. We need to exit if there was an error.
+      process.exit(1);
```

Comment:
isn't this..dangerous? handle the error instead of exiting. also, the end result is the same anyway: the process will exit if the error bubbles up to the process without being handled.

Reply:
Nope, this needs to be handled here. The fact is that the supported Clay ecosystem doesn't support the `sites` index not being populated. If you don't exit here then you'll still be able to spin up everything else but Kiln won't work properly. 

If the implementation wants to capture the exit and handle it differently, it can, but from a platform perspective this should be a hard failure.

Reply:
i still don't think you end to use process.exit here.

> Throwing an error has a similar effect as calling process.exit() in that code execution in this function stops immediately. However, calling functions have the opportunity to catch the error and respond to it in a graceful manner. If there is no intervention up the call stack, then the uncaughtException event is fired on process. If there are no event handlers, then Node.js will fire the exit event and exit with a non-zero exit code just like when process.exit() is called; if there are event handlers, then it is up to you to manually call process.exit() to specify the exit code to use.

The key is that throwing an error gives the application an opportunity to catch the error and recover from it, which is almost always the desired case when dealing with module code.

source: https://stackoverflow.com/questions/24149784/do-we-need-process-exit1-when-we-use-promises-in-node-js

Reply:
My main issue is that this truly is a fatal error to the platform when using the supported configuration and needs to be fixed by stopping the server and restarting with the proper configuration. The application can still start if we do not hard exit here, but it won't be usable. 

Also, we don't want to throw an Error because the aim of this PR is to pipe all messaging through the `clay-log` module. If we throw an error then the format of messages for tools like ELK will be broken and more tooling will be broken. 

An implementation can still recover from exiting the process, there are event to listen to and clustering which allow for recovery. For example, if populating the `sites` index failed because there was an interruption in the connection between the Elastic instance and the Clay instance, the process would exit but the clustering would start a brand new process to try and recover. This would repeat until the connection is restored, and then it would work normally.

Reply:
if the application exits, why would the elastic restore be useful to us? even the documentation advises against it: ```Rather than calling process.exit() directly, the code should set the process.exitCode and allow the process to exit naturally by avoiding scheduling any additional work for the event loop:```

https://nodejs.org/api/process.html#process_process_exit_code

since @nelsonpecora already approved this, maybe he can chime in here.